README.md

```diff
@@ -201,6 +201,38 @@ Note that `#count` is calculated automatically based on the number of batches in
 your collection, and your Task’s progress will be displayed in terms of batches
 (not the number of records in the relation).
 
+On databases with large tables, sometime using an `ActiveRecord::Relation` to define a collection will
+cause your queries to timeout. One pattern to to have performant tasks is to look up the range of IDs that
+need to be operated on, then create a collection of ranges to query your models. For example:
+
+```ruby
+module Maintenance
+  class BackfillALotOfPosts < MaintenanceTasks::Task
+    # Generates an array like
+    # [(1..1_000_000),(1_000_001..2_000_000),(2_000_001..3_000_000), ...]
+    def collection
+      ranges = []
+      100.times do |i|
+        range_start = (1_000_000 * i) + 1
+        range_end = 1_000_000 * (i + 1)
+        ranges << (range_start..range_end)
+      end
+      ranges
+    end
+
+    def process(range)
+      Post
+        .where(id: range, some_column: "Some Value")
+        .in_batches { |batch| batch.update_all(some_column: "Some New Value") }
```

Comment:
Would it be more reasonable to recommend building smaller ranges instead of using `in_batches` inside the `process` method? It will make the `collection` method slower but will allow the task to be interrupted more frequently and reduce the risk of the `process` method taking more than the max configured time for a single iteration

Reply:
Wouldn't that result in a lot more queries, thus be slower?

Not a DB expert by any means, but my understanding is that `SELECT` is generally going to be a lot faster than `UPDATE`/`DELETE`. 

Using an example of a job that needs to iterate over 100M rows, the approach here would do 100 `SELECT` queries, and somewhere between 0-1000 `UPDATE`/`DELETE` queries per each SELECT. In practice on my last run, this was normally <200 `DELETE` queries per `SELECT`. Back of the envelope math:

>100 `SELECT` * 200 `DELETE`/`UPDATE` = 20,000 total `DELETE`/`UPDATE` queries

If I did that in batches over, say, 100M rows, that would be 100,000 `DELETE` queries, which are inherently slower.

Maybe I'm wrong on that. If I am, I'd be happy to update the example. Though, either way, I'd hope people will use this as inspiration and write whatever is best for their app.